dis_012_landslide_hazard_alerts/contents/src/__init__.py

```diff
@@ -8,225 +8,348 @@
+            time_field:  optional, name of column that will store datetime information (string)
+    RETURN  list of existing IDs in the table, pulled from the id_field column (list of strings)
+    '''
+    # check it the table already exists in Carto
+    if cartosql.tableExists(table, user=CARTO_USER, key=CARTO_KEY):
+        # if the table does exist, get a list of all the values in the id_field column
+        logging.info('Fetching existing IDs')
+        r = cartosql.getFields(id_field, table, f='csv', post=True, user=CARTO_USER, key=CARTO_KEY)
+        # turn the response into a list of strings, removing the first and last entries (header and an empty space at end)
+        return r.text.split('\r\n')[1:-1]
+    else:
+        # if the table does not exist, create it with columns based on the schema input
+        logging.info('Table {} does not exist, creating'.format(table))
+        cartosql.createTable(table, schema, user=CARTO_USER, key=CARTO_KEY)
+        # if a unique ID field is specified, set it as a unique index in the Carto table; when you upload data, Carto
+        # will ensure no two rows have the same entry in this column and return an error if you try to upload a row with
+        # a duplicate unique ID
+        if id_field:
+            cartosql.createIndex(table, id_field, unique=True, user=CARTO_USER, key=CARTO_KEY)
+        # if a time_field is specified, set it as an index in the Carto table; this is not a unique index
+        if time_field:
+            cartosql.createIndex(table, time_field, user=CARTO_USER, key=CARTO_KEY)
+        # return an empty list because there are no IDs in the new table yet
+        return []
+
+'''
+FUNCTIONS FOR THIS DATASET
+
+The functions below have been tailored to this specific dataset.
+They should all be checked because their format likely will need to be changed.
+'''
+
 def genUID(datetime, position_in_geojson):
-    '''Generate unique id'''
+    '''Generate unique id using date and index in retrieved json
+    INPUT   datetime: date for which we want to generate id (string)
+            position_in_geojson: index of the datetime in json (string)
+    '''
     return '{}_{}'.format(datetime, position_in_geojson)
 
-def processData(exclude_ids, table, src_url):
+def processData(src_url, table, existing_ids):
+    '''
+    Fetch, process and upload new data
+    INPUT   src_url: url where you can find the download link for the source data (string)
+            table: name of table in Carto where we will upload the data (string)
+            existing_ids: list of date IDs that we already have in our Carto table (list of strings)
+    RETURN  new_ids: list of unique ids of new data sent to Carto table (list of strings)
+    '''
+    # create an empty list to store unique ids of new data we will be sending to Carto table
     new_ids = []
-
+    # set the start date for collecting data to oldest date of data stored in the Carto
+    # table in the format (YYYY-MM-DD)
```

Comment:
Here's an actual example of a start date: '2019-05-20T15:01:31.450342' so it includes time too, but I don't know how to abbreviate microseconds, so i'm just gonna leave off the comment about the format
```suggestion
```